src/main/java/site/coach_coach/coach_coach_server/routine/service/RoutineService.java

```diff
@@ -161,4 +162,21 @@ public void validateAndDeleteRoutine(Long routineId, Long userIdByJwt) {
 
 		routineRepository.deleteById(routineId);
 	}
+
+	@Transactional
+	public Routine validateAccessToRoutine(Long routineId, Long userIdByJwt) {
+		Routine routine = routineRepository.findById(routineId)
+			.orElseThrow(() -> new NoExistRoutineException(ErrorMessage.NOT_FOUND_ROUTINE));
+
+		if (!routine.getUserId().equals(userIdByJwt)) {
+			Long coachId = getCoachId(userIdByJwt);
+			if (routine.getCoachId() != coachId) {
+				throw new AccessDeniedException();
+			}
+		} else if (routine.getCoachId() != null) {
+			throw new AccessDeniedException();
+		}
+
+		return routine;
+	}
```

Comment:
```suggestion
	public void checkAccessToRoutine(Long routineId, Long userIdByJwt) {
        Routine routine = routineRepository.findById(routineId)
            .orElseThrow(() -> new NoExistRoutineException(ErrorMessage.NOT_FOUND_ROUTINE));

        if (routine.getUserId().equals(userIdByJwt)) {
            return;
        }

        Long coachId = routineService.getCoachId(userIdByJwt);
        if (routine.getCoachId() != null && routine.getCoachId().equals(coachId)) {
            return;
        }

        throw new AccessDeniedException();
}
```
이전 pr에서 validate 가독성 높이는 게 쉽지 않다고 하셨는데, 이렇게 작성하면 안 되나요?? 제가 테스트해봤을 땐.. 음 되는 것 같긴해서..

Reply:
코치가 만들어준 루틴의 경우, 일반회원이 카테고리 삭제나 수정이 불가능합니다.
그래서 처음 if문 안에 coachId 비교를 넣었습니다.
한비님이 생각하시는 코드로는 위와 같은 예외 처리가 힘들거 같아요..!

Reply:
아아아아 그렇네요.. 본인꺼를 못바꾼다는 걸.. 까먹었네요 알았습니당